index.js

```diff
@@ -22,12 +22,27 @@ async function run() {
   // Create a map of changeset entries
   const entryMap = prepareChangesetEntryMap(changesetEntries, prNumber, prLink);
 
+  // Check if the "skip" option is present in the changeset entries
+  const skipLabel = "skip-changelog";
   if (entryMap["skip"]) {
     if (Object.keys(entryMap).length > 1) {
       throw new CategoryWithSkipOptionError();
     } else {
       console.log("No changeset file created or updated.");
-      return;
+      try {
+        // Add the "skip-changelog" label to the PR
+        await updatePRLabel(owner, repo, prNumber, skipLabel, true);
+        return;
+      } catch (error) {
+        console.error(`Error updating label "${skipLabel}" for PR #${prNumber}: ${error.message}`);
+      }
+    }
+  } else {
+    try {
+      // Check if the "skip-changelog" label is present on the PR and remove it
+      await updatePRLabel(owner, repo, prNumber, skipLabel, false);
+    } catch (error) {
+      console.error(`Error updating label "${skipLabel}" for PR #${prNumber}: ${error.message}`);
```

Comment:
Why do we need this else statement? We would never have this case. Or is it to avoid the case when someone manually adds that label in the repo? Also, I feel you are repeating yourself. Code in index.js has to be concise. Here a suggestion where we can introduce modularization

```js
const skipLabel = "skip-changelog";

async function handleSkipLabel(entryMap) {
    if (entryMap["skip"]) {
        if (Object.keys(entryMap).length > 1) {
            throw new CategoryWithSkipOptionError();
        }
        await updateLabelAndLog(true);
    } else {
        await updateLabelAndLog(false);
    }
}

async function updateLabelAndLog(addLabel) {
    try {
        await updatePRLabel(owner, repo, prNumber, skipLabel, addLabel);
        if (addLabel) {
            console.log("No changeset file created or updated.");
        }
    } catch (error) {
        console.error(`Error updating label "${skipLabel}" for PR #${prNumber}: ${error.message}`);
    }
}

// Usage
handleSkipLabel(entryMap);
```

Reply:
Admittedly, it would be rare that we would encounter an edge case like this. However, when I was testing the feature, I started by changing our test PR description from a description that had category entries to one that just had a `- skip` entry. I got the following error in the workflow logs:

```
Run BigSamu/OpenSearch_Change_Set_Create_Action@main
Extracting data for PR #2 in BigSamu/OpenSearch-Dashboards
Found 1 changelog entry.
Error updating label "skip-changelog" for PR #2: Label does not exist
Error updating label "skip-changelog" for PR #2: Label does not exist
File: changelogs/fragments/2.yml updated successfully.
```
Additionally, since it's good practice in general to check if something exists before removing it, I figured I would add the `else` statement as a precaution.

Reply:
Also, I like your suggestion for modularizing the code in `index.js`. I can work on implementing that.

Reply:
@JohnathonBowers, This suggestion was already implemented or not? Don't see this updated in this PR.

Reply:
> @JohnathonBowers, This suggestion was already implemented or not? Don't see this updated in this PR.

Yes, I implemented this suggestion in PR #11 .

Reply:
The modularization that @BigSamu mentioned has been implemented in PR #11 .